state: support `state:ignore`

cathay4t opened the pull request:
Introducing `InterfaceState.IGNORE` for:
    * The ignored interface will not be changed when applying desire state.
    * The state verification will not be impacted by ignored interface.

Test cases added.

ffmancera commented:
I am fine with the current approach but I feel like it could be confusing to use `state` to mark the interface as ignore. Because `ignore` it is not a state itself. Maybe it would be better to add a new parameter that could be `mark:ignore` or something like that, indeed that could be useful to mark the interfaces in the future, what do you think?

EdDev commented:
> I am fine with the current approach but I feel like it could be confusing to use `state` to mark the interface as ignore. Because `ignore` it is not a state itself. Maybe it would be better to add a new parameter that could be `mark:ignore` or something like that, indeed that could be useful to mark the interfaces in the future, what do you think?

Adding another field is odd. I think that when applying, "ignore" means... "do not do UP, DOWN or DISABLE" which translates to not touching it. So it seems ok to me.

When reporting, it will never be IGNORE, right? This merits a comment in the commit, on the API and in documentation.

cathay4t commented:
> > I am fine with the current approach but I feel like it could be confusing to use `state` to mark the interface as ignore. Because `ignore` it is not a state itself. Maybe it would be better to add a new parameter that could be `mark:ignore` or something like that, indeed that could be useful to mark the interfaces in the future, what do you think?
> 
> Adding another field is odd. I think that when applying, "ignore" means... "do not do UP, DOWN or DISABLE" which translates to not touching it. So it seems ok to me.
> 
> When reporting, it will never be IGNORE, right? This merits a comment in the commit, on the API and in documentation.

The ignore is only used when applying desire state.
The use case is RHV/oVirt is expecting nmstate/NM don't touch the TUN/TAP interface managed by libvirt.

Currently, nmstate don't touch them as they are unknown to us. But I think the correct way is mark them as `ignored` in desire state.

ffmancera reviewed (approved):
As everyone agrees on introducing `state: ignore` I will not block on this. Approved, thanks!

cathay4t commented:
QE approval request sent. Please wait test feedback before merge it.

cathay4t committed:
state: support `state:ignore`

Introducing `InterfaceState.IGNORE` for:
    * The ignored interface will not be changed when applying desire state.
    * The state verification will not be impacted by ignored interface.

This state will be only used for `libnmstate.apply()`.

Test cases added.

Signed-off-by: Gris Ge <fge@redhat.com>

smyle10 commented:
Applied 'state: ignore' to main port and subordinate port respectively, it works well.

cathay4t commented:
Timeout on waiting comsumer QE. Merging.
